src/main/kotlin/org/rust/ide/surroundWith/expression/RustExpressionSurrounderBase.kt

```diff
@@ -1,15 +1,19 @@
-package org.rust.ide.surroundWith
+package org.rust.ide.surroundWith.expression
 
+import com.intellij.codeInsight.CodeInsightUtilCore
 import com.intellij.lang.surroundWith.Surrounder
 import com.intellij.openapi.editor.Editor
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.TextRange
 import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustElementFactory
 import org.rust.lang.core.psi.RustExprElement
 
-abstract class RustExpressionSurrounderBase : Surrounder {
+abstract class RustExpressionSurrounderBase<E: RustExprElement> : Surrounder {
+    abstract fun createTemplate(project: Project): E
+    abstract fun getLeafExpression(expression: E) : RustExprElement
```

Comment:
Whitespace around `:` is a bit off, and there is an unused import. **Reformat code** action should take care of these.